usr/sbin/p11sak/p11sak.c

```diff
@@ -1319,12 +1531,21 @@ static CK_RV pub_key_print_attributes(CK_SESSION_HANDLE session,
                         CK_BBOOL2a(*(CK_BBOOL*) bool_tmplt[i].pValue));
             }
         }
-        printf("|\n");
+        rc = print_vendor(session, hkey, 1);
+        if (rc != CKR_OK) {
+            return rc;
+        }
     } else {
         /* Short print */
         printf(" |");
         for (i = 2; i < KEY_MAX_BOOL_ATTR_COUNT; i++)
             short_print(i, bool_tmplt, kt_PUBLIC);
+        rc = print_vendor(session, hkey, 0);
+        if (rc != CKR_OK) {
+            printf("Attribute retrieval failed (error code 0x%lX: %s)\n", rc,
```

Comment:
Error messages normally go to stderr.  Is the output to stdout desired?

Reply:
No, fixed